common/workunit/workunit.cpp

```diff
@@ -7056,14 +7062,27 @@ wuTokenStates verifyWorkunitDAToken(const char * distributedAccessToken)
     }
 
     //Verify user matches
-    if (!isEmptyString(cw->queryUser()) || !isEmptyString(tokUser.str()))
+    bool wuUserExist = !isEmptyString(cw->queryUser());
+    bool tokUserExist = !isEmptyString(tokUser.str());
+    if (wuUserExist && tokUserExist)
     {
-        if (!streq(cw->queryUser(), tokUser.str()))
+        //if both users are found, they must match
+        if (!streq(cw->queryUser(), tokUser.str()) || !streq(cw->queryUser(), ctxUser))
         {
-            ERRLOG("verifyWorkunitDAToken : token user does not match workunit");
+            ERRLOG("verifyWorkunitDAToken : token user (%s) does not match WU user (%s) or context user (%s)", tokUser.str(), cw->queryUser(), ctxUser);
             return wuTokenInvalid;
         }
     }
+    else if (!wuUserExist && !tokUserExist)//both users will be empty if no security enabled
+    {
+        //both users will be empty if no security enabled
```

Comment:
this should probably return false if ctxUser is not empty - only possible to occur on a system started without security, but security then enabled.

Reply:
This method is not called if there is no security manager